tools/apigee/apigee.py

```diff
@@ -105,10 +113,17 @@ async def create_developer(self, email: str, first_name: str,
         return email
       raise hse
 
-  async def get_dc_api_keys(self, project_id: str) -> list[str]:
+  async def batch_get_dc_api_keys(
+      self, project_ids: list[str]) -> dict[str, list[str]]:
+    futures = [self.get_dc_api_keys(project_id) for project_id in project_ids]
+    keys = dict(await asyncio.gather(*futures))
+    return keys
+
+  async def get_dc_api_keys(self, project_id: str) -> tuple[str, list[str]]:
     key_names = await self.fetch_dc_api_key_names(project_id)
     futures = [self.fetch_dc_api_key_string(key_name) for key_name in key_names]
-    return list(filter(lambda x: x, await asyncio.gather(*futures)))
+    keys = list(filter(lambda x: x, await asyncio.gather(*futures)))
```

Comment:
Any risk that this results in enough concurrent requests to cause quota issues?

Reply:
It is guarded by the semaphore so we're ok.